modules/dashboard/jsx/dashboardIndex.js

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Loader from 'Loader';
+import WelcomePanel from './panels/welcome';
+import RecruitmentPanel from './panels/recruitment';
+import StudyProgressionPanel from './panels/studyprogression';
+import MyTasksPanel from './panels/mytasks';
+import DocumentRepositoryPanel from './panels/documentrepository';
+import BehaviouralFeedbackPanel from './panels/behaviouralfeedback';
+
+/**
+ * Dashboard.
+ *
+ * @description the dashboard of LORIS.
+ *
+ * @author Alizée Wickenheiser
+ * @version 1.0.0
+ *
+ */
+class Dashboard extends React.Component {
+  /**
+   * Constructor of component
+   * @param {object} props - the component properties.
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      display: {
+        panel: {
+          welcome: true,
+          recruitment: true,
+          studyProgression: true,
+          myTasks: true,
+          documentRepository: true,
+          behaviouralFeedback: true,
+        },
+      },
+      data: null,
+    };
+    // Bind component instance to custom methods
+    this.fetchInitializerData = this.fetchInitializerData.bind(this);
+  }
+
+  /**
+   * Executes after component mounts.
+   */
+  componentDidMount() {
+    this.fetchInitializerData();
+  }
+
+  /**
+   * Used with sending POST data to the server.
+   * @param {object} json - json object converted for POST.
+   * @return {string} send in POST to server.
+   */
+  urlSearchParams(json) {
+    return Object.keys(json).map((key) => {
+      return encodeURIComponent(key) + '=' + encodeURIComponent(json[key]);
+    }).join('&');
+  }
+
+  /**
+   * Retrieve data from the dataURL and save it in state.
+   */
+  fetchInitializerData() {
+    const send = this.urlSearchParams({
+      command: 'initialize',
+    });
+    fetch(
+      this.props.dataURL, {
+          method: 'POST',
+          mode: 'same-origin',
+          credentials: 'include',
+          redirect: 'follow',
+          headers: {
+            'Accept': 'application/json, text/plain, */*',
+            'Content-Type': 'application/x-www-form-urlencoded',
+          },
+          body: send,
+        }
+    ).then((response) => response.json())
+        .then(
+            (data) => {
+              this.setState({
+                data: data,
+                isLoaded: true,
+              });
+            })
+        .catch(
```

Comment:
This will only catch network error and TypeError from `response.json()` on line#81 when an html page is returned (which happen with \NDB_Page subclasses).
It should, instead check if `response.ok` is false and throw an Error object. Note that the only place where the error message will be shown is in the console.